tmt/hardware.py

```diff
@@ -635,6 +630,80 @@ def variants(
         yield (members or []) + [self]
 
 
+class SizedConstraint(Constraint['Size']):
```

Comment:
Could we rather name it `SizeConstraint`, feels inconsistent with other constraints naming.